game-app/game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -124,7 +124,8 @@ public class MapData {
   private Set<String> undrawnTerritoriesNames;
   private final Map<Image, List<Point>> decorations = new HashMap<>();
   private final Map<String, Image> territoryNameImages = new HashMap<>();
-  private final Map<String, Image> effectImages = new HashMap<>();
+  // Use a synchronized map since getTerritoryEffectImage() is called from multiple threads.
+  private final Map<String, Image> effectImages = Collections.synchronizedMap(new HashMap<>());
```

Comment:
If the key is always guaranteed to be non-null you might want to consider using `ConcurrentHashMap` instead which yields better performance in most cases

Reply:
Done,